tests/distributions/test_flows.py

```diff
@@ -125,11 +129,22 @@ def test_batchnorm_jacobians(self):
         for input_dim in [2, 3, 5, 7, 9, 11]:
             self._test_jacobian(input_dim, self._make_batchnorm)
 
+    def test_blocknaf_jacobians(self):
+        for activation in ['ELU', 'LeakyReLU', 'sigmoid', 'tanh']:
+            for residual in [None, 'normal', 'gated']:
+                for input_dim in [2, 3, 5, 7, 9, 11]:
```

Comment:
do we need quite so many `input_dim`s...? unfortuantely, we always have somewhat limited travis cores

more generally, we probably don't need every last cross product

Reply:
Fixed!

Reply:
I've reduced the number of tests